wagtail/contrib/settings/forms.py

```diff
@@ -5,17 +5,8 @@
 from wagtail.admin.staticfiles import versioned_static
 from wagtail.models import Site
 
-
 class SiteSwitchForm(forms.Form):
-    site = forms.ChoiceField(choices=[])
-
-    @property
-    def media(self):
-        return forms.Media(
-            js=[
-                versioned_static("wagtailsettings/js/site-switcher.js"),
-            ]
-        )
+    site = forms.ChoiceField(choices=[], widget=forms.Select(attrs={'data-auto-submit': ''}))
 
     def __init__(self, current_site, model, **kwargs):
         initial_data = {"site": self.get_change_url(current_site, model)}
```

Comment:
Note: There will still need to be a solution for ensuring that the 'current' site is disabled or somehow set to not trigger a page reload.

Maybe that logic could be in the controller, but ideally we can find a more semantic way to do this in the HTML.

Reply:
As per other comment, this ended up not being required.